From 40d37c8d0643575c6191419a2637e49133fd321d Mon Sep 17 00:00:00 2001
From: Sveinar Søpler <cybermax@dexter.no>
Date: Sat, 30 Nov 2019 14:27:18 +0100
Subject: [PATCH] cybwine

---
 dlls/kernel32/process.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/dlls/kernel32/process.c b/dlls/kernel32/process.c
index e37e26e..749603c 100644
--- a/dlls/kernel32/process.c
+++ b/dlls/kernel32/process.c
@@ -192,11 +192,11 @@ void WINAPI start_process( LPTHREAD_START_ROUTINE entry, PEB *peb )
     {
         if (CreateEventA(0, 0, 0, "__winestaging_warn_event") && GetLastError() != ERROR_ALREADY_EXISTS)
         {
-            FIXME_(winediag)("Wine Staging %s is a testing version containing experimental patches.\n", wine_get_version());
-            FIXME_(winediag)("Please mention your exact version when filing bug reports on winehq.org.\n");
+            FIXME_(winediag)("Cybwine %s is a testing version containing experimental patches.\n", wine_get_version());
+            FIXME_(winediag)("Please don't report bugs about it on winehq.org and use your brain!\n");
         }
         else
-            WARN_(winediag)("Wine Staging %s is a testing version containing experimental patches.\n", wine_get_version());
+            WARN_(winediag)("Cybwine %s is a testing version containing experimental patches.\n", wine_get_version());
 
 
         if (!CheckRemoteDebuggerPresent( GetCurrentProcess(), &being_debugged ))
-- 
2.17.1

