---
 dlls/dxgi/adapter.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dlls/dxgi/adapter.c b/dlls/dxgi/adapter.c
index b96904e90c0..522a3aedf2d 100644
--- a/dlls/dxgi/adapter.c
+++ b/dlls/dxgi/adapter.c
@@ -334,7 +334,7 @@
             return E_INVALIDARG;
     }
 
-    return hr;
+    return S_OK;
 }
 
 static HRESULT STDMETHODCALLTYPE dxgi_adapter_SetVideoMemoryReservation(IWineDXGIAdapter *iface,
-- 
2.18.1
