--- a/dlls/wined3d/context.c	2018-11-24 21:13:02.981072542 +0100
+++ b/dlls/wined3d/context.c	2018-11-25 13:18:46.343026331 +0100
@@ -3803,6 +3803,7 @@
         const struct wined3d_state *state, enum wined3d_shader_type shader_type)
 {
     unsigned int bind_idx, shader_sampler_count, base, count, i;
+    const struct wined3d_gl_info *gl_info = context->gl_info;
     const struct wined3d_device *device = context->device;
     struct wined3d_shader_sampler_map_entry *entry;
     struct wined3d_shader_resource_view *view;
@@ -3832,6 +3833,10 @@
         if (!(view = state->shader_resource_view[shader_type][entry->resource_idx]))
         {
             WARN("No resource view bound at index %u, %u.\n", shader_type, entry->resource_idx);
+            context_active_texture(context, gl_info, bind_idx);
+            context_bind_texture(context, GL_NONE, 0);
+            GL_EXTCALL(glBindSampler(bind_idx, 0));
+            checkGLcall("unbind shader resource");
             continue;
         }
 
